x/tally/keeper/abci.go

```diff
@@ -200,8 +200,16 @@ func (k Keeper) FilterAndTally(ctx sdk.Context, req types.Request) (tallyvm.VmRe
 		"arguments", args,
 	)
 	vmRes := tallyvm.ExecuteTallyVm(tallyWasm.Bytecode, args, map[string]string{
-		"VM_MODE":   "tally",
-		"CONSENSUS": fmt.Sprintf("%v", consensus),
+		"VM_MODE":               "tally",
+		"CONSENSUS":             fmt.Sprintf("%v", consensus),
+		"DR_ID":                 req.ID,
+		"DR_INPUT":              req.DrInputs,
+		"BINARY_ID":             req.DrBinaryID,
+		"DR_REPLICATION_FACTOR": fmt.Sprintf("%v", req.ReplicationFactor),
+		"DR_GAS_PRICE":          req.GasPrice,
+		"DR_GAS_LIMIT":          req.GasLimit,
+		"DR_MEMO":               req.Memo,
+		"DR_PAYBACK_ADDRESS":    req.PaybackAddress,
```

Comment:
I can't really tell from this, but are these hex encoded instead of base64?

Reply:
base64. In fact, the Sudo call to the contract fails if it's in hex or something other than base64 encoding

Reply:
Ah Im just saying that the contract assumes it to be a base64 string. So if VM wants it to be hex-encoded, we will have to convert it on chain.